examples/multiple_data_source_examples/public/components/home.tsx

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { EuiPage, EuiPageSideBar, EuiSideNav } from '@elastic/eui';
+import { BrowserRouter as Router, Route, withRouter, RouteComponentProps } from 'react-router-dom';
+import { AppMountContext, CoreStart, MountPoint } from 'opensearch-dashboards/public';
+import { DataSourceManagementPluginSetup } from 'src/plugins/data_source_management/public';
+import { NavigationPublicPluginStart } from 'src/plugins/navigation/public';
+import { DataSourceListActiveExample } from './data_source_list_active_example';
+import { DataSourceListAllExample } from './data_source_list_all_example';
+import { DataSourceMultiSelectableExample } from './data_source_multi_selectable_example';
+import { DataSourceSelectableExample } from './data_source_selectable_example';
+import { DataSourceSelectorExample } from './data_source_selector_example';
+import { DataSourceViaTopNavMenuExample } from './data_source_via_top_nav_menu';
+import { DataSourceViewExample } from './data_source_view_example';
+
+export interface HomeProps {
+  basename: string;
+  notifications: CoreStart['notifications'];
+  http: CoreStart['http'];
+  savedObjects: CoreStart['savedObjects'];
+  dataSourceEnabled: boolean;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+  navigateToApp: CoreStart['application']['navigateToApp'];
+  navigation: NavigationPublicPluginStart;
+  setActionMenu?: (menuMount: MountPoint | undefined) => void;
+}
+
+interface PageDef {
+  title: string;
+  id: string;
+  component: React.ReactNode;
+}
+
+type NavProps = RouteComponentProps & {
+  navigateToApp: AppMountContext['core']['application']['navigateToApp'];
+  pages: PageDef[];
+};
+
+const Nav = withRouter(({ history, pages }: NavProps) => {
```

Comment:
if use Route component, probably don't need `withRouter` here

Reply:
withRouter is different from Router component https://v5.reactrouter.com/web/api/withRouter